Cython/Compiler/Options.py

```diff
@@ -385,6 +385,20 @@ class DEFER_ANALYSIS_OF_ARGUMENTS:
 }
 
 
+# a list of directives that (when used as a decorator) are only applied to
+# the object they decorate and not to its children.
+immediate_decorator_directives = (
+    'cfunc', 'ccall', 'cclass',
+    # function signature directives
+    'inline', 'exceptval', 'returns',
+    # class directives
+    'freelist', 'no_gc', 'no_gc_clear', 'type_version_tag', 'final',
+    'auto_pickle', 'internal',
+    # testing directives
+    'test_fail_if_path_exists', 'test_assert_path_exists',
+)
```

Comment:
Given how this is used, it should rather be a set.
```suggestion
immediate_decorator_directives = {
    'cfunc', 'ccall', 'cclass',
    # function signature directives
    'inline', 'exceptval', 'returns',
    # class directives
    'freelist', 'no_gc', 'no_gc_clear', 'type_version_tag', 'final',
    'auto_pickle', 'internal',
    # testing directives
    'test_fail_if_path_exists', 'test_assert_path_exists',
}
```